routes/cats.js

```diff
@@ -18,6 +18,34 @@ router.get('/', async (req, res) => {
     }
 });
 
+// @route   GET api/cats/:id
+// @desc    Get one cat
+// @access  Public
+router.get('/:id', async (req, res) => {
+    try {
+        let cat = await Cat.findById(req.params.id);
+
+        res.json(cat);
+    } catch (err) {
+        console.error(err.message);
+        res.status(500).send('Server Error');
+    }
+});
+
+// @route   GET  api/cats/search/:search
+// @desc    Get results from search bar
+// @access  Public
+router.get('/search/:search', async (req, res) => {
+    try {
+
+        let search = await Cat.find({name: new RegExp(req.params.search, "ig")});
```

Comment:
Good job,
You can probably search for more than just name in this function in the future